allauth/account/views.py

```diff
@@ -628,13 +628,7 @@ def dispatch(self, request, *args, **kwargs):
         return super(PasswordChangeView, self).dispatch(request, *args, **kwargs)
 
     def render_to_response(self, context, **response_kwargs):
-        if self.request.user.is_anonymous:
-            # We end up here when `ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True`.
-            redirect_url = get_adapter(self.request).get_logout_redirect_url(
-                self.request
-            )
-            return HttpResponseRedirect(redirect_url)
-        elif not self.request.user.has_usable_password():
+        if not self.request.user.has_usable_password():
```

Comment:
Why did you alter this code? AFAIK, Django's `AnonymousUser` does not have a `has_usable_password()` method.

Reply:
AFAIK the comment _We end up here when `ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True`_ is not valid (this is why I'm adding a test), which means : 
 - if we end up on this view as anonymous, we should be redirected somewhere, LOGIN_URL maybe ?
 - we're connected and if we submit this view and successfully change the password:
   - we have LOGOUT_ON_PASSWORD_CHANGE to True and should be logged out
   - LOGOUT_ON_PASSWORD_CHANGE is False and we just redirect to success_url